templates/funcs.go

```diff
@@ -255,6 +284,10 @@ func render(f pgs.Field) string {
 			s = bb.String()
 		}
 	} else if t.IsEmbed() {
+		if obsType == zap.ObfuscationType_STARS {
+			return fmt.Sprintf("\no.AddString(\"%s\", \"***\")\n", name(f))
```

Comment:
Im not 100% sure you can `return` because there is further processing of `s` below

Reply:
There's not a single `s +=` in the code

All of `s =` could be replaced by a `return` and lots of `else` could then be removed

Provided that the order of the conditions is correct ofc

Reply:
Solved that